skins/cat17/faq-page/src/components/Question.vue

```diff
@@ -1,12 +1,44 @@
 <template>
 <div id="question" class="space-above">
-	<p class="title">{{ content }}</p>
+	<div v-bind:class="[ isOpen ? 'border-secondary' : 'border-primary' ]">
+		<div class="inline-icon clickable" @click="toggle()">
+			<h5 v-bind:class="[ isOpen ? 'secondary-color' : 'title-primary-color' ]">
+				{{ content.question }}
+			</h5>
+			<i v-bind:class="[ isOpen ? 'icon-keyboard_arrow_up secondary-color' : 'icon-keyboard_arrow_down primary-color' ]"
+			   data-content-target="/page/Häufige Fragen"
+			   data-track-content
+			   data-content-name="Toggle expand/collapse"
+			   data-content-piece="Toggle expand/collapse">
+			</i>
+		</div>
+		<div v-show="isOpen"  v-html="content.visible_text"></div>
+	</div>
 </div>
 </template>
 
 <script>
 export default {
 	name: 'question',
-	props: [ 'content' ]
+	props: [ 'content', 'visibleQuestionId' ],
+	data() {
+		return {
+			collapse: true
+		};
+	},
+	computed: {
+		isOpen: function () {
+			return this.$vnode.key === this.visibleQuestionId;
+		}
+	},
+	methods: {
+		toggle: function () {
+			if ( !this.isOpen ) {
+				this.$emit( 'question-opened', this.$vnode.key );
+			} else {
+				this.$emit( 'question-opened' );
```

Comment:
Oooh, that is so clever! Emitting the event without id will close the question, because an empty id won't match the key in `isOpen`. Great way to do things. 

It took me a while to figure that out, though. Not sure if that makes it too clever and how to mitigate - passing an empty string or adding an inline comment perhaps? But not a blocker, you can also leave the code like it is.

Reply:
I also thought about passing empty string because the prop the parent sends to the child is an empty string initially. But.. `key` is `Number` so we emit the same type of event with different type of data. I thought that's not super correct, but I do agree with the argument that it's not clear what's going on in the `else`. I'll add a comment for now and if in 6 months from now we find ourselves wondering what does this do, we can change it.